File: branches/2.5.0.2/iRINGTools.ESBServices/iRINGTools.Common/src/org/iringtools/widgets/grid/Field.java
package org.iringtools.widgets.grid;

import java.io.Serializable;

public class Field implements Serializable
{
  private static final long serialVersionUID = 1L;
  
  protected String name;  // header
  protected String dataIndex;
  protected String type;  // data type
  protected int width;
  protected boolean fixed = false;  // fixed width
  protected boolean filterable = true;
  protected boolean sortable = true;
  protected int editMode = 0;
  
  public void setName(String name)
  {
    this.name = name;
  }
  
  public String getName()
  {
    return name;
  }
  
  public void setDataIndex(String dataIndex)
  {
    this.dataIndex = dataIndex;
  }

  public String getDataIndex()
  {
    return dataIndex;
  }

  public void setType(String type)
  {
    this.type = type;
  }
  
  public String getType()
  {
    return type;
  } 
  
  public void setWidth(int width)
  {
    this.width = width;
  }
  
  public int getWidth()
  {
    return width;
  } 
  
  public void setFixed(boolean fixed)
  {
    this.fixed = fixed;
  }
  
  public boolean getFixed()
  {
    return fixed;
  }

  public void setFilterable(boolean filterable)
  {
    this.filterable = filterable;
  }

  public boolean getFilterable()
  {
    return filterable;
  }

  public void setSortable(boolean sortable)
  {
    this.sortable = sortable;
  }

  public boolean getSortable()
  {
    return sortable;
  } 
  
  public void setEditMode(int editMode)
  {
    this.editMode = editMode;
  }
  
  public int getEditMode()
  {
    return editMode;
  } 
}

File: Libraries/iRINGLibrary/Desktop/Field.cs
using System.Runtime.Serialization;
using System.ComponentModel;

namespace org.iringtools.library
{
	[DataContract(Name = "Field", Namespace = "http://www.iringtools.org/library")]
	public class Field
	{
		[DataMember(Name = "name", IsRequired = true, Order = 0)]
		public string name { get; set; }  // header

		[DataMember(Name = "dataIndex", IsRequired = true, Order = 1)]
		public string dataIndex { get; set; }
    
		[DataMember(Name = "type", IsRequired = true, Order = 2)]
		public string type { get; set; }  // data type

		[DataMember(Name = "width", IsRequired = true, Order = 3)]
		public int width { get; set; }
    
    [DataMember(Name = "fixed", IsRequired = true, Order = 4)]
    public bool @fixed { get; set; }

		[DataMember(Name = "filterable", IsRequired = true, Order = 5)]
    public bool filterable { get; set; }

		[DataMember(Name = "sortable", IsRequired = true, Order = 6)]
    public bool sortable { get; set; }

    [DataMember(Name = "keytype", IsRequired = true, Order = 7)]
    public string keytype { get; set; }

    [DataMember(Name = "hidden", EmitDefaultValue = false, Order = 8)]
    public bool hidden { get; set; }
	}
}